src/main/java/hudson/plugins/git/GitChangeSet.java

```diff
@@ -371,11 +387,26 @@ public User findOrCreateUser(String csAuthor, String csAuthorEmail, boolean crea
 
             if (user == null) {
                 try {
-                    user = User.get(csAuthorEmail, true);
-                    user.setFullName(csAuthor);
-                    if (hasHudsonTasksMailer())
-                        setMail(user, csAuthorEmail);
-                    user.save();
+                    user = User.get(csAuthorEmail, !useExistingAccountWithSameEmail);
+                    boolean setUserDetails = true;
+                    if (user == null && useExistingAccountWithSameEmail && hasHudsonTasksMailer()) {
+                        for(User existingUser : User.getAll()) {
+                            if (getMail(existingUser).equalsIgnoreCase(csAuthorEmail)) {
```

Comment:
Won't this throw a null pointer exception in the two branches where `getMail(user)` returns null?

What safeguards are there to prevent the null pointer exception?

Reply:
you are right.
lines 382-384 however guarantee that csAuthorEmail is not null at this point, so I switched to csAuthorEmail.equalsIgnoreCase(getMail(existingUser)